jgrapht-core/src/main/java/org/jgrapht/experimental/dag/DirectedAcyclicGraph.java

```diff
@@ -532,6 +533,82 @@ private void reorder(Set<V> df, Set<V> db, Visited visited)
     }
 
     /**
+     * This function is implemented recursively. Therefore only graphs without
+     * any cycles are allowed.
+     *
+     * @param graph graph to look for ancestors in.
+     * @param vertex the vertex to get the ancestors of.
+     *
+     * @return {@link List} of ancestors of the vertex in the given graph.
```

Comment:
since the ancestors are unique (and not guaranteed to be in any particular order), the return type should be Set, not List